src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -22,20 +23,23 @@
     private final Date date;
     private final Amount amount;
     private final Direction direction;
+    private final Account account;
     private final Description description;
     private final Set<Category> categories = new HashSet<>();
 
     /**
      * Every field must be present and not null.
+     * Since we only accept one category in the argument, it is added to the set of categories.
      */
     public Transaction(Date date, Amount amount, Direction direction, Description description,
-                       Set<Category> categories) {
-        requireAllNonNull(date, amount, direction, description, categories);
+                       Account account, Category category) {
+        requireAllNonNull(date, amount, direction, category);
         this.date = date;
         this.amount = amount;
         this.direction = direction;
+        this.account = account;
         this.description = description;
-        this.categories.addAll(categories);
+        this.categories.add(category);
```

Comment:
Perhaps we could still accept multiple categories in the argument.

Reply:
I've adjusted the Transactions constructor to accept varargs for categories, but this will result in q a few changes to the Add command, i'll make another PR after those changes. Thanks!

Reply:
Hmm I think the main thing is that we have to figure out how to parse multiple categories.

Reply:
You can reference Address Book's implementation for tags. `Person` has `Set<Tag>` , `ParserUtil` has `parseTag(String tag)` and `parseTags(Collection<String> tags)` . But yea, future PR.

Reply:
Sure, i'll take reference from those